cumulus/parachains/integration-tests/emulated/tests/coretime/coretime-rococo/src/tests/coretime_interface.rs

```diff
@@ -0,0 +1,223 @@
+				) => {},
+			]
+		);
+	});
+
+	// Check that the request_core_count message was processed successfully. This will fail if the
+	// weights are misconfigured.
+	Rococo::execute_with(|| {
+		Rococo::assert_ump_queue_processed(true, Some(CoretimeRococo::para_id()), None);
+
+		assert_expected_events!(
+			Rococo,
+			vec![
+				RelayEvent::MessageQueue(
+					pallet_message_queue::Event::Processed { success: true, .. }
+				) => {},
+			]
+		);
+	});
+
+	// Keep track of the relay chain block number so we can fast forward while still checking the
+	// right block.
+	let mut block_number_cursor = Rococo::ext_wrapper(<Rococo as Chain>::System::block_number);
+
+	let config = CoretimeRococo::ext_wrapper(|| {
+		Configuration::<<CoretimeRococo as Chain>::Runtime>::get()
+			.expect("Pallet was configured earlier.")
+	});
+
+	// Now run up to the block before the sale is rotated.
+	while block_number_cursor < TIMESLICE_PERIOD - config.advance_notice - 1 {
+		CoretimeRococo::execute_with(|| {
+			// Hooks don't run in emulated tests - workaround.
+			<CoretimeRococo as CoretimeRococoPallet>::Broker::on_initialize(
+				<CoretimeRococo as Chain>::System::block_number(),
+			);
+		});
+
+		Rococo::ext_wrapper(|| {
+			block_number_cursor = <Rococo as Chain>::System::block_number();
+		});
+	}
+
+	// In this block we trigger assign core.
+	CoretimeRococo::execute_with(|| {
+		// Hooks don't run in emulated tests - workaround.
+		<CoretimeRococo as CoretimeRococoPallet>::Broker::on_initialize(
+			<CoretimeRococo as Chain>::System::block_number(),
+		);
+
+		assert_expected_events!(
+			CoretimeRococo,
+			vec![
+				CoretimeEvent::Broker(
+					pallet_broker::Event::SaleInitialized { .. }
+				) => {},
+				CoretimeEvent::Broker(
+					pallet_broker::Event::CoreAssigned { .. }
+				) => {},
+				CoretimeEvent::ParachainSystem(
+					cumulus_pallet_parachain_system::Event::UpwardMessageSent { .. }
+				) => {},
+			]
+		);
+	});
+
+	// In this block we trigger request revenue.
+	CoretimeRococo::execute_with(|| {
+		// Hooks don't run in emulated tests - workaround.
+		<CoretimeRococo as CoretimeRococoPallet>::Broker::on_initialize(
+			<CoretimeRococo as Chain>::System::block_number(),
+		);
+
+		assert_expected_events!(
+			CoretimeRococo,
+			vec![
+				CoretimeEvent::ParachainSystem(
+					cumulus_pallet_parachain_system::Event::UpwardMessageSent { .. }
+				) => {},
+			]
+		);
+	});
+
+	// Check that the assign_core and request_revenue_info_at messages were processed successfully.
+	// This will fail if the weights are misconfigured.
+	Rococo::execute_with(|| {
+		Rococo::assert_ump_queue_processed(true, Some(CoretimeRococo::para_id()), None);
+
+		assert_expected_events!(
+			Rococo,
+			vec![
+				RelayEvent::MessageQueue(
+					pallet_message_queue::Event::Processed { success: true, .. }
+				) => {},
+				RelayEvent::MessageQueue(
+					pallet_message_queue::Event::Processed { success: true, .. }
+				) => {},
```

Comment:
minor, but we probably can trigger Rococo between two CoretimeRococo above, to process these two messages one by one. I think that would be closer to what happens in prod.

Reply:
Good point, done in c4ab8443ca0